### PR TITLE
Add TPU support to the CI

### PR DESCRIPTION
**Pull request by krshrimali**
## What does this PR do?

We are in a process to add support for TPU Training and Validation on multiple cores/single core in Lightning Flash. This PR acts in the direction.

**What this does?** Adds TPU testing _support to the CI_, confirms with adding a _very simple_ test.

**What it won't add?** Add TPU tests for Flash. A separate PR will be made to add those tests, and discuss the design.

## Before submitting
- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? [not needed for typos/docs]
- [ ] Did you verify **new and existing tests pass** locally with your changes?
- [ ] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by kaushikb11 — APPROVED** [review]
Thanks Kush! ❤️

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,6 +1,25 @@
...
+      - "release/*"
+      - "refs/tags/*"
+
+pr:
+  - "master"
+  - "release/*"
+
 references:
 
   make_docs: &make_docs
```
Comment (Borda) [inline .circleci/config.yml:25]: lets remove building docs :)

File: dockers/tpu-tests/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# limitations under the License.
+
+ARG PYTHON_VERSION=3.9
+ARG PYTORCH_VERSION=1.9
+
+FROM pytorchlightning/pytorch_lightning:base-xla-py${PYTHON_VERSION}-torch${PYTORCH_VERSION}
+
+LABEL maintainer="Lightning-AI <https://github.com/Lightning-AI>"
+
+COPY ./ ./pytorch-lightning/
```
Comment (Borda) [inline dockers/tpu-tests/Dockerfile:22]: you shall be copying Flash, right?

File: dockers/tpu-tests/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+# Pull the legacy checkpoints
+RUN cd pytorch-lightning && \
```
Comment (Borda) [inline dockers/tpu-tests/Dockerfile:25]: ALso not related

Comment (krshrimali) [reply dockers/tpu-tests/Dockerfile:25]: Agreed, I'll remove this after looking into your suggestion [here](https://github.com/Lightning-AI/lightning-flash/pull/1433#discussion_r958408745).

File: .circleci/config.yml (hunk 1 above)
Comment (krshrimali) [reply .circleci/config.yml:25]: Sure! I'll remove with the next commit.

File: dockers/tpu-tests/Dockerfile (hunk 2 above)
Comment (Borda) [reply dockers/tpu-tests/Dockerfile:22]: I can help :)

Comment (krshrimali) [reply dockers/tpu-tests/Dockerfile]: TPU CI is passing! 😄 Does it look good to you, @Borda?

**Comment by krshrimali** [comment]
TPU CI is passing! 😄 Mind taking another look cc: @Borda?

**Review by Borda — APPROVED** [review]

**Pull request merged**